### PR TITLE
Upload file to drive code

### PR DESCRIPTION
**Pull request by jyokush**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,2 @@
-# vba-migrator-addin
+## Upload file to drive
+Code itself is self-explanatory
```
Comment (monikadhok) [inline README.md]: This is not encouraged. Please explain what this code does in brief and you can point to the link from where you have got the code pointers.

File: uploadFileToDrive.vb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        'Creating upload Request
+        Dim UploadRequest As FilesResource.InsertMediaUpload = Service.Files.Insert(TheFile, Stream, TheFile.MimeType)
+        UploadRequest.SupportsTeamDrives = True
+        UploadRequest.Fields = "id"
+        'Call to upload
+        UploadRequest.Upload()
+        'Retrieving the response body to check our file is successfully uploaded or not
+        Dim file As File = UploadRequest.ResponseBody
+        If file Is Nothing Then
+            MsgBox("Something went wrong!! I am null body, So i can't able to upload the project to the drive")
```
Comment (monikadhok) [inline uploadFileToDrive.vb]: Please write a proper message.

File: uploadFileToDrive.vb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            Return ""
+        Else
+            MsgBox("Upload Finished")
```
Comment (monikadhok) [inline uploadFileToDrive.vb]: We can get rid of this message box.

File: uploadFileToDrive.vb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        Dim MyUserCredential As UserCredential =
+        GoogleWebAuthorizationBroker.AuthorizeAsync(New ClientSecrets() With
+            {.ClientId = ClientId, .ClientSecret = ClientSecret},
+            {DriveService.Scope.Drive}, "user", CancellationToken.None).Result
+        Service = New DriveService(New BaseClientService.Initializer() With {.HttpClientInitializer = MyUserCredential,
+            .ApplicationName = "Google Sheets Compatibility Report"})
+    End Sub
+
+
+        'Fuction to upload the file to drive using google drive api version-2.
```
Comment (monikadhok) [inline uploadFileToDrive.vb]: Please indent the line properly.

File: uploadFileToDrive.vb [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    Public Function UploadFile(FilePath As String) As String
+        'Set the application name whatever you have entered during enabling the drive api.
+        If Service.ApplicationName <> "Google Sheets Compatibility Report" Then CreateService()
```
Comment (monikadhok) [inline uploadFileToDrive.vb]: Create constants for the string.

File: uploadFileToDrive.vb [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        'Define parameters of request.
+        Dim TheFile As Google.Apis.Drive.v2.Data.File =
+            New Google.Apis.Drive.v2.Data.File()
+        'Write the name of the file by which you want to save  
+        TheFile.Title = "internDemo2"
```
Comment (monikadhok) [inline uploadFileToDrive.vb]: Please use constants for strings in the code.

File: uploadFileToDrive.vb [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        'Write whatever you want to write as description of your file
```
Comment (monikadhok) [inline uploadFileToDrive.vb]: Please use proper comments.

**Review by monikadhok — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jyokush) [reply README.md]: Done

File: uploadFileToDrive.vb (hunk 2 above)
Comment (jyokush) [reply uploadFileToDrive.vb]: Actually here request-response body is null.Done.

File: uploadFileToDrive.vb (hunk 3 above)
Comment (jyokush) [reply uploadFileToDrive.vb]: Okay, Done.

File: uploadFileToDrive.vb (hunk 4 above)
Comment (jyokush) [reply uploadFileToDrive.vb]: Done.

File: uploadFileToDrive.vb (hunk 6 above)
Comment (jyokush) [reply uploadFileToDrive.vb]: Done.

File: uploadFileToDrive.vb (hunk 7 above)
Comment (jyokush) [reply uploadFileToDrive.vb]: Tried.

File: uploadFileToDrive.vb [hunk 8]
```diff
@@ -3,48 +3,33 @@ Imports Google.Apis.Auth.OAuth2
...
 Imports Google.Apis.Drive.v2.Data
-''' <summary>
-'''For this you need to have "google account" & "visual studio 2013 or later"
-''' To import the above library firstly we have to install the "Google.Apis.Drive.v2" 
-''' in the project using NuGet Package manager console "Install-Package Google.Apis.Drive.v2".
-''' Then enable the google api for your google account to get your own "ClientId" & 
-''' "ClientSecretId", using this link "https://console.developers.google.com/flows/enableapi?apiid=drive"
-''' and you can follow this video to enable api "https://www.youtube.com/watch?v=xtqpWG5KDXY&t=1s"
-''' To upload the file to Drive, "UploadFile("file path")" member function of this class
-''' have to call by passing the path of the file to upload.
```
Comment (monikadhok) [inline uploadFileToDrive.vb]: Brief summary

File: uploadFileToDrive.vb [hunk 9]
```diff
@@ -3,48 +3,33 @@ Imports Google.Apis.Auth.OAuth2
...
 Public Class uploadFileToDrive
     Private Service As DriveService = New DriveService()
     'Create Drive API service.
     Private Sub CreateService()
-        'Set the client id and clientSecret id you get after anabling the google drive api
-        'for your google account.
-        Dim ClientId =
-            "601010958158-ri1h9bipsbkfjip0qjhcnatfhdupnn08.apps.googleusercontent.com"
+        'Set your "clientId" and "clientSecretId".
+        Dim ClientId = "601010958158-ri1h9bipsbkfjip0qjhcnatfhdupnn08.apps.googleusercontent.com"
```
Comment (monikadhok) [inline uploadFileToDrive.vb]: Parameterize this.

**Review by monikadhok — CHANGES_REQUESTED** [review]

File: uploadFileToDrive.vb (hunk 9 above)
Comment (jyokush) [reply uploadFileToDrive.vb]: Done.

File: uploadFileToDrive.vb (hunk 8 above)
Comment (jyokush) [reply uploadFileToDrive.vb]: Done.

**Review by monikadhok — DISMISSED** [review]

**Review by monikadhok — APPROVED** [review]

**Pull request merged**